odoo/tools/misc.py

```diff
@@ -1013,19 +1013,34 @@ def popmap(self):
 class OrderedSet(MutableSet):
     """ A set collection that remembers the elements first insertion order. """
     __slots__ = ['_map']
+
     def __init__(self, elems=()):
         self._map = OrderedDict((elem, None) for elem in elems)
+
     def __contains__(self, elem):
         return elem in self._map
+
     def __iter__(self):
         return iter(self._map)
+
     def __len__(self):
         return len(self._map)
+
     def add(self, elem):
         self._map[elem] = None
+
     def discard(self, elem):
         self._map.pop(elem, None)
 
+    def update(self, elems):
+        for elem in elems:
+            self.add(elem)
+
+    def __str__(self):
+        return self.__repr__()
+
+    def __repr__(self):
+        return 'OrderedSet([%s])' % ', '.join(repr(elem) for elem in self)
```

Comment:
Can use f-strings. Also one of the rare occasions where `map` makes perfect sense.

Alternatively as I doubt anyone's going to be perf-limited on this:
```suggestion
        return f'OrderedSet({list(self)!r})'
```

Reply:
This repr is only for debug purpose so no, perf are not at all an issue.